hls4ml/templates/vivado/nnet_utils/nnet_recurrent.h

```diff
@@ -0,0 +1,851 @@
+    typename CONFIG_T::V_t     param_V[CONFIG_T::n_state][CONFIG_T::n_out])
+{
+
+    // Initialize the state variable -- will maintain state between function calls
+    //static typename CONFIG_T::state_t newstate[CONFIG_T::n_state];
+    std::cout << "Pre-State: [ "; for (int ii = 0; ii < CONFIG_T::n_state; ii++) std::cout << newstate[ii] << " "; std::cout << "]" << std::endl;
+
+    // Operation: U*input
+    data_T inputcache;
+    typename CONFIG_T::state_t inputmult[CONFIG_T::n_in][CONFIG_T::n_state];
+    typename CONFIG_T::state_t inputacc[CONFIG_T::n_state];
+    InputProd1: for(int ii = 0; ii < CONFIG_T::n_in; ii++) {
+        inputcache = data[ii];
+        InputProd2: for(int jj = 0; jj < CONFIG_T::n_state; jj++) {
+            inputmult[ii][jj] = inputcache * param_U[ii][jj];
+        }
+    }
+    for(int iacc = 0; iacc < CONFIG_T::n_state; iacc++) {
+        inputacc[iacc] = 0;
+    }
+    InputAccum1: for(int ii = 0; ii < CONFIG_T::n_in; ii++) {
+        InputAccum2: for(int jj = 0; jj < CONFIG_T::n_state; jj++) {
+            inputacc[jj] += inputmult[ii][jj];
+        }
+    }
+
+    // Operation: W*state
+    data_T statecache;
+    typename CONFIG_T::state_t statemult[CONFIG_T::n_state][CONFIG_T::n_state];
+    typename CONFIG_T::state_t stateacc[CONFIG_T::n_state];
+    StateProd1: for(int ii = 0; ii < CONFIG_T::n_state; ii++) {
+        statecache = newstate[ii];
+        StateProd2: for(int jj = 0; jj < CONFIG_T::n_state; jj++) {
+            statemult[ii][jj] = statecache * param_W[ii][jj];
+        }
+    }
+    for(int iacc = 0; iacc < CONFIG_T::n_state; iacc++) {
+        stateacc[iacc] = 0;
+    }
+    StateAccum1: for(int ii = 0; ii < CONFIG_T::n_state; ii++) {
+        StateAccum2: for(int jj = 0; jj < CONFIG_T::n_state; jj++) {
+            stateacc[jj] += statemult[ii][jj];
+        }
+    }
+
+    // Operation: U*input + W*state
+    typename CONFIG_T::state_t rawstate[CONFIG_T::n_state];
+    for(int iacc = 0; iacc < CONFIG_T::n_state; iacc++) {
+        rawstate[iacc] = inputacc[iacc] + stateacc[iacc];
+    }
+
+    std::cout << "Post-State: [ "; for (int ii = 0; ii < CONFIG_T::n_state; ii++) std::cout << rawstate[ii] << " "; std::cout << "]" << std::endl;
+
+    // Run activation function
+    if (CONFIG_T::activation_type == activ_relu){
+        relu<typename CONFIG_T::state_t, typename CONFIG_T::state_t, ACT_CONFIG_T>(rawstate, newstate);
+    }
+    else if (CONFIG_T::activation_type == activ_sigmoid){
+      sigmoid<typename CONFIG_T::state_t, typename CONFIG_T::state_t, ACT_CONFIG_T>(rawstate, newstate);
+    }
+    else if (CONFIG_T::activation_type == activ_tanh){
+        tanh<typename CONFIG_T::state_t, typename CONFIG_T::state_t, ACT_CONFIG_T>(rawstate, newstate);
+    }
+
+    std::cout << "Activated State: [ "; for (int ii = 0; ii < CONFIG_T::n_state; ii++) std::cout << newstate[ii] << " "; std::cout << "]" << std::endl;
```

Comment:
This is not useful unless we are debugging the implementation itself so it should be removed.

Reply:
This was part of the basic RNN layer. Will be removing those functions as they are not used anyway.